doc/designs/dtlsv1_3/dtlsv1_3-main.md

```diff
@@ -70,6 +70,26 @@ section 5.9.
 The DTLSv1.3 implementation modifies the epoch according to RFC9147 section 6.1
 for DTLSv1.3 connections.
 
+#### DTLS ACK records (RFC9147 Section 7)
+
+ACK's are sent for KeyUpdates, NewSessionTicket and Finish (client).
```

Comment:
ACKs and post-handshake messages are subtle. Do you correctly handle the case when you're requested to send another message (e.g. key_update_requested or caller asking to send a ticket) while you're in the middle of sending something? Lots of things can flow in parallel post-handshake and the document is very ill-specified.

Reply:
What do you mean by "sending something"? Is it protected records? Or are you concerned with KeyUpdates crossing in flight? Please give an example.

Reply:
I think this will require tests are setup in order to make sure that we correctly handle all corner cases. I'll note this in the documentation and then revisit later.

Reply:
Thanks for taking the time btw!